### DISCUSSION TITLE
Why Prisma should be inserted to GraphQL `context`

### DISCUSSION THREAD

**Discussion by LauraBeatris** [prisma/prisma#11209] [Q&A] [answered]
Hey! 👋🏻

I've been reading about [query optimization](https://www.prisma.io/docs/guides/performance-and-optimization/query-optimization-performance#solving-n1-in-graphql-with-findunique-and-prismas-dataloader) and the examples there are accessing a Prisma instance from `context` argument 

Also, the following is mentioned:
> Automatic batching of findUnique is particularly useful in a GraphQL context. GraphQL runs a separate resolver function for every field, which can make it difficult to optimize a nested query.
> the Prisma dataloader ✔ batches the findUnique queries.

Currently, I'm calling Prisma in functions declared outside of the resolver:
```js
export const getPostAuthor = ({ postId }: { postId: string }) => (
  dbClient.post.findUnique({
    where: {
      id: postId
    }
  }).author()
)
```
So if it's connected to the context, would batching still be handled by Prisma?

**Comment by LauraBeatris** [comment, marked as answer]
Even by not adding a Prisma instance to context, I think the batching still working based on the following:
```graphql
  users(take: $take) {
    id
    name
    posts {
      title
    }
  }
```

<img width="714" alt="Screenshot 2022-01-17 at 13 01 37" src="https://user-images.githubusercontent.com/48022589/149766222-ee3eee81-3e76-4cd9-a81f-7d521948e2ee.png">

There's only 1 query log for posts, considering that there are 4 users with > 1 post each

**Reply by austincrim** [reply]
Hey @LauraBeatris 👋 ,

Batching will always work as long as you are using the same Prisma Client instance. Passing in a client instance via context is an easy and clean way to make sure this happens, but it's totally up to you.

Let me know if I can clarify that further!

**Reply by LauraBeatris** [reply]
@austincrim Makes sense! Thanks for explaining 😁

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
